components/browser/icons/src/main/java/mozilla/components/browser/icons/processor/ColorProcessor.kt

```diff
@@ -0,0 +1,87 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.icons.processor
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.Color.rgb
+import androidx.annotation.ColorInt
+import androidx.palette.graphics.Palette
+import mozilla.components.browser.icons.Icon
+import mozilla.components.browser.icons.IconRequest
+import kotlin.math.max
+
+/**
+ * [IconProcessor] implementation to extract the dominant color from the icon.
+ */
+class ColorProcessor : IconProcessor {
+
+    override fun process(context: Context, request: IconRequest, resource: IconRequest.Resource?, icon: Icon): Icon {
+        if (icon.bitmap == null || icon.color != null) return icon
+
+        getEdgeColor(icon.bitmap)?.let { edgeColor ->
+            return icon.copy(color = edgeColor)
+        }
```

Comment:
I'm confused why we're using this, given that we're using `Palette` now?

Reply:
I ported this existing code from Fennec, but I agree it makes more sense to just always use `Palette`.

Reply:
I remember this was added for activity stream - where we used the color as background color for the tile on which we draw the icon. If we pick a color that is used along the edges then the icon "merges" with the background.